optapy-core/src/main/python/optaplanner_java_interop.py

```diff
@@ -563,16 +575,18 @@ def _generate_planning_entity_class(python_class: Type, annotation_data: Dict[st
     global unique_class_id
     ensure_init()
     from org.optaplanner.optapy import PythonWrapperGenerator  # noqa
+    class_identifier = _get_class_identifier_for_object(python_class)
     optaplanner_annotations = _get_optaplanner_annotations(python_class)
     parent_class = None
     has_eq_and_hashcode = _does_class_define_eq_or_hashcode(python_class)
     if len(python_class.__bases__) == 1 and hasattr(python_class.__bases__[0], '__optapy_java_class'):
         parent_class = get_class(python_class.__bases__[0])
-    out = PythonWrapperGenerator.definePlanningEntityClass(python_class.__name__ + str(unique_class_id),
+    out = PythonWrapperGenerator.definePlanningEntityClass(f'{class_identifier}{unique_class_id}',
```

Comment:
This string concatenation repeats all over the file; maybe it's worth making it a method. The same applies to generating the `unique_class_id`. Although the algorithm is just an increment, it may change in the future.